### PR TITLE
#136 - The SQLite Storage Provider is creating duplicate experiments

### LINKED ISSUE #136: The SQLite Storage Provider is creating duplicate experiments
**Describe the bug**
If we use the SQLite storage provider, it will create duplicate experiments if one already exists. The expected behavior is that it will just return existing one. We should add an integration test to assert this.

**To Reproduce**
1. Call `CreateExperimentAsync("test")` twice

**Expected behavior**
No duplicate experiments (with the same name) should be created. Instead, we should return the ID of the existing experiment.

The current implementation is as follows, and we need to change it to check if the experiment already exists.

```
        public async Task<Guid> CreateExperimentAsync(string name)
        {
            using (var db = new LocalDbContext())
            {
                var experiment = new Experiment(name);
                await db.Experiments.AddAsync(experiment);
                await db.SaveChangesAsync();
                
                return experiment.Id;
            }
        }
```

### PR DESCRIPTION
**Pull request by mbrdev**
## Resolves
Fixes #136 

## Description
Added a new test for the issue and implemented the duplicate check on CreateExperimentAsync for SQLiteMetaDataStore.

### CODE REVIEW

**Comment by mbrdev** [comment]
@aslotte, not sure why the tests are failing with "Login failed for user 'sa'." error as I haven't changed the connection string etc.

**Comment by aslotte** [comment]
@mbrdev - no worries, I've been monitoring this issue and I'm trying to understand what may be up. Thanks for this contribution, let me have a look at the PR!

File: test/MLOps.NET.SQLite.IntegrationTests/SQLiteMetaDataStoreTests.cs [hunk 1]
```diff
@@ -136,6 +136,23 @@ public async Task CreateRunAsync_WithoutGitCommitHash_ShouldProvideEmptyGitCommi
...
+        {
+            // Arrange
+            var sut = new MLOpsBuilder()
+	            .UseSQLite()
+	            .UseModelRepository(new Mock<IModelRepository>().Object)
+	            .Build();
+
+            //Act
+            var experimentId = await sut.LifeCycle.CreateExperimentAsync("test");
+	        var experimentId2 = await sut.LifeCycle.CreateExperimentAsync("test");
```
Comment (aslotte) [inline test/MLOps.NET.SQLite.IntegrationTests/SQLiteMetaDataStoreTests.cs:150]: Just a minor comment here about the indentation being one or two spaces too many

File: src/MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs [hunk 2]
```diff
@@ -16,11 +16,17 @@ public async Task<Guid> CreateExperimentAsync(string name)
...
             using (var db = new LocalDbContext())
             {
-                var experiment = new Experiment(name);
-                await db.Experiments.AddAsync(experiment);
-                await db.SaveChangesAsync();
-                
-                return experiment.Id;
+	            var existingExperiment = db.Experiments.FirstOrDefault(x => x.ExperimentName == name);
+
+	            if (existingExperiment == null)
```
Comment (aslotte) [inline src/MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs:21]: Just a minor comment here about the indentation being one or two spaces too many

File: test/MLOps.NET.SQLite.IntegrationTests/SQLiteMetaDataStoreTests.cs (hunk 1 above)
Comment (mbrdev) [reply test/MLOps.NET.SQLite.IntegrationTests/SQLiteMetaDataStoreTests.cs]: Should be fixed now

File: src/MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs (hunk 2 above)
Comment (mbrdev) [reply src/MLOps.NET.SQLite/Storage/SQLiteMetaDataStore.cs]: Should be fixed now

**Comment by aslotte** [comment]
Perfecto. Thank you very much sir!

**Pull request merged**